src/main/kotlin/kweb/state/render.kt

```diff
@@ -21,50 +20,55 @@ private val logger = KotlinLogging.logger {}
 
 fun <T : Any?> ElementCreator<*>.render(
     value: KVal<T>,
-    container: ElementCreator<*>.() -> Element
-        = { span().setAttribute("style", JsonPrimitive("display: contents;")) },
     block: ElementCreator<Element>.(T) -> Unit
-) {
-
-    // NOTE: Per https://github.com/kwebio/kweb-core/issues/151 eventually render() won't rely on a container element.
-    val containerElement: Element = container(this)
+) : RenderFragment {
 
     val previousElementCreator: AtomicReference<ElementCreator<Element>?> = AtomicReference(null)
 
     val renderState = AtomicReference(NOT_RENDERING)
 
+    //TODO this could possibly be improved
+    val renderFragment: RenderFragment = if (parent.browser.isCatchingOutbound() == null) {
+        parent.browser.batch(WebBrowser.CatcherType.RENDER) {
+            val startSpan = span().classes("RenderMarkerStart")
+            val endSpan = span().classes("RenderMarkerEnd")
+            RenderFragment(startSpan.id, endSpan.id)
+        }
+    } else {
+        val startSpan = span().classes("RenderMarkerStart")
```

Comment:
Nitpick but we should probably put these strings in a `private var renderMarkerStart = "RenderMarkerStart"` if they're referenced more than once.

We'll probably want to use a shorter string at some point too for efficiency.